wikidata/client.py

```diff
@@ -177,20 +178,28 @@ def decode_datavalue(self,
                       self.datavalue_decoder)
         return decode(self, datatype, datavalue)
 
-    def request(self, path: str) -> Union[
+    def request(self, path: str) -> Optional[Union[
         bool, int, float, str,
         Mapping[
             str,
             Union[bool, int, float, str, Mapping[str, object], Sequence]
         ],
         Sequence[Union[bool, int, float, str, Mapping[str, object], Sequence]]
-    ]:
+    ]]:
         logger = logging.getLogger(__name__ + '.Client.request')
         url = urllib.parse.urljoin(self.base_url, path)
         result = self.cache_policy.get(CacheKey(url))
         if result is None:
             logger.debug('%r: no cache; make a request...', url)
-            response = self.opener.open(url)
+            try:
+                response = self.opener.open(url)
+            except urllib.error.HTTPError as e:
+                logger.debug('HTTP error code: ', e.code)
```

Comment:
For the sake of debug, it could log a traceback as well.  The `exc_info=True` option purposes that.  [(See also the docs.)][1]

[1]: https://docs.python.org/3/library/logging.html#logging.Logger.debug

Reply:
I added the argument.